### PR TITLE
Implement main_query parameter in get_meta_query

### LINKED ISSUE #22442: Price Filter Bug - affecting manually added shortcodes
**Describe the bug**

The price filter widget is also filtering manually added product shortcodes rather than the just the result set.

**To Reproduce**
1. Set up the Storefront theme.
2. Add a single product shortcode widget to the sidebar. e.g. [product id="37"]
3. Filter by price. Set the min price to be greater than the product whose id is 37.
4. The product shortcode disappears from the sidebar.

**Screenshots**
Live link with the issue - [https://themedemo.commercegurus.com/storefront-testing/shop/
](url)

**Expected behavior**
The result set should filter - but the manually added shortcode in the sidebar should not disappear.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by rrennick**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

The `$main_query` parameter of `WC_Query::get_meta_query` was not implemented within the function. Both the product shortcode and the Products by Rating used `get_meta_query()` with the default value of `false`. This PR adds the logic to exclude the price meta query filter when `$main_query` is `false` and the product query is not the main query.

Closes #22442 .

### How to test the changes in this Pull Request:

1. Add a product shortcode to a text widget
2. Add the products by rating widget to the sidebar
3. Add the price filter widget to the sidebar
4. Use the price filter to exclude the prices of the highest rated product and the product in the shortcode

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### Changelog entry

> Restrict the price filter widget to filtering the main product query.

### CODE REVIEW

File: includes/class-wc-query.php [hunk 1]
```diff
@@ -598,10 +598,14 @@ public function order_by_popularity_post_clauses( $args ) {
 	 * @return array
 	 */
 	public function get_meta_query( $meta_query = array(), $main_query = false ) {
+		global $wp_query;
+
 		if ( ! is_array( $meta_query ) ) {
 			$meta_query = array();
 		}
-		$meta_query['price_filter'] = $this->price_filter_meta_query();
+		if ( $main_query || $wp_query === self::$product_query ) {
```
Comment (mikejolley) [inline includes/class-wc-query.php]: This comparison is not needed - $main_query is passed. The functions calling this method take care of main query checks.

**Review by mikejolley — CHANGES_REQUESTED** [review]

File: includes/class-wc-query.php (hunk 1 above)
Comment (rrennick) [reply includes/class-wc-query.php]: Thanks @mikejolley . WP query check removed.

Comment (mikejolley) [reply includes/class-wc-query.php]: Don't forget the global :)

Comment (rrennick) [reply includes/class-wc-query.php]: Thanks :) Removed.

**Review by mikejolley — APPROVED** [review]

**Pull request merged**